src/AutocompleteBadges.js

```diff
@@ -0,0 +1,142 @@
+'use strict';
+
+import templates from './AutocompleteBadges.soy.js';
+import Autocomplete from 'metal-autocomplete';
+import Component from 'metal-component';
+import Soy from 'metal-soy';
+
+class AutocompleteBadges extends Component {
+
+	/**
+	 * @inheritDoc
+	 */
+	attached() {
+		this.autocomplete_ = new Autocomplete({
+			inputElement: this.refs.input,
+			data: this.getFilteredElements_.bind(this)
+		});		
+
+		this.autocomplete_.on('select', this.onListItemSelected_.bind(this));
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	rendered() {
+		if(this.refs.listBadges) {
```

Comment:
There are a few minor formatting issues like this in the code (we always use a space between the **if** and the parentheses, for example). We have an automated tool that fixes those to conform with our standards though, so if you just run `gulp format` on this repo you'll get those fixed automatically for you :)